extensions/time-traveling-world-clock/src/index.tsx

```diff
@@ -0,0 +1,225 @@
+        }}
+        onAction={() => setOffsetHrs((o) => o + 1)}
+      />
+      <Action
+        title="-1hr"
+        shortcut={{
+          modifiers: ["cmd"],
+          key: "arrowLeft",
+        }}
+        onAction={() => setOffsetHrs((o) => o - 1)}
+      />
+      <Action
+        title="Clear hours offset"
+        shortcut={{
+          modifiers: ["cmd"],
+          key: "0",
+        }}
+        onAction={() => setOffsetHrs(() => 0)}
+      />
+      <Action.Push
+        title="Add a city"
+        shortcut={{
+          modifiers: ["cmd", "shift"],
+          key: "=",
+        }}
+        target={
+          <AddCityView
+            onAdd={(c) => {
+              const _cities = sortCitiesByTimeZone(cities.concat(c));
+              setCities(_cities);
+              setJSON("cities", _cities);
+            }}
+          />
+        }
+      />
+      {city ? (
+        <Action
+          title="Remove city"
+          style={Action.Style.Destructive}
+          shortcut={{
+            modifiers: ["cmd"],
+            key: "delete",
+          }}
+          onAction={() => {
+            const _cities = cities.filter((_city) => _city !== city);
+            setCities(_cities);
+            setJSON("cities", _cities);
+          }}
+        />
+      ) : null}
+    </ActionPanel>
+  );
+
+  const localDate = new Date();
+  localDate.setTime(localDate.getTime() + offsetHrs * 60 * 60 * 1000);
+  const localTime = timeFormatter.format(localDate);
+  const localDateString = dateFormatter.format(localDate);
+  const todayDateString = dateFormatter.format(new Date());
+
+  const subtitle = localTime + " " + (offsetHrs && localDateString !== todayDateString ? localDateString : "");
+
+  const local = (
+    <List.Item
+      key={"."}
+      title={"Local time"}
+      subtitle={subtitle}
+      icon={getIconForTime(localDate)}
+      actions={actions({ time: localTime })}
+    />
+  );
+
+  return (
+    <List filtering searchBarPlaceholder="Adjust hour with ⌘← and ⌘→">
+      {local}
+      {cities.map((c) => {
+        const _date = new Date();
+        _date.setTime(_date.getTime() + offsetHrs * 60 * 60 * 1000);
+        const date = new Date(_date.toLocaleString("en-US", { timeZone: c.timezone }));
+        const timeString = timeFormatter.format(date);
+        const dateString = dateFormatter.format(date);
+        const subtitle = `${timeString} ${todayDateString === dateString ? "" : dateString}`;
+
+        return (
+          <List.Item
+            key={c.city}
+            title={c.city}
+            subtitle={subtitle}
+            icon={getIconForTime(date)}
+            actions={actions({ time: subtitle, city: c })}
+          />
+        );
+      })}
+    </List>
+  );
+}
+
+function getIconForTime(date: Date) {
+  const h = date.getHours();
+
+  if (h >= 6 && h < 10) {
+    return "icon-morning.png";
+  }
+
+  if (h >= 10 && h < 17) {
+    return "icon-day.png";
+  }
+
+  if (h >= 17 && h < 20) {
```

Comment:
Maybe it's up for debate but I'd argue that `17` is still the day 😬 

Isn't there a way to know the sunrise/sunset hours for each city to return more accurate icons?